wallet/src/de/schildbach/wallet/ui/invite/InviteHandler.kt

```diff
@@ -157,40 +134,52 @@ class InviteHandler(val activity: FragmentActivity, private val analytics: Analy
     }
 
     private fun showInvalidInviteDialog(displayName: String) {
-        val title = activity.getString(R.string.invitation_invalid_invite_title)
-        val message = activity.getString(R.string.invitation_invalid_invite_message, displayName)
-        val inviteErrorDialog = FancyAlertDialog.newInstance(title, message, R.drawable.ic_invalid_invite, R.string.okay, 0)
-        inviteErrorDialog.show(activity.supportFragmentManager, null)
-        handleDialogResult(activity)
+        AdaptiveDialog.create(
+            R.drawable.ic_invalid_invite,
+            activity.getString(R.string.invitation_invalid_invite_title),
+            activity.getString(R.string.invitation_invalid_invite_message, displayName),
+            activity.getString(R.string.okay)
+        ).show(activity) {
+            handleDialogButtonClick(activity)
+        }
         analytics.logEvent(AnalyticsConstants.Invites.ERROR_INVALID, bundleOf())
     }
 
     fun showUsernameAlreadyDialog() {
-        val inviteErrorDialog = FancyAlertDialog.newInstance(
-            R.string.invitation_username_already_found_title,
-            R.string.invitation_username_already_found_message,
-            R.drawable.ic_invalid_invite, R.string.okay, 0
-        )
-        inviteErrorDialog.show(activity.supportFragmentManager, null)
-        handleDialogResult(activity)
+        AdaptiveDialog.create(
+            R.drawable.ic_invalid_invite,
+            activity.getString(R.string.invitation_username_already_found_title),
+            activity.getString(R.string.invitation_username_already_found_message),
+            activity.getString(R.string.okay)
+        ).show(activity) {
+            handleDialogButtonClick(activity)
+        }
         analytics.logEvent(AnalyticsConstants.Invites.ERROR_USERNAME_TAKEN, bundleOf())
     }
 
     private fun showInviteAlreadyClaimedDialog(invite: InvitationLinkData) {
         val inviteAlreadyClaimedDialog = InviteAlreadyClaimedDialog.newInstance(activity, invite)
+        inviteAlreadyClaimedDialog.onFancyAlertButtonsClickListener = object :
+            FancyAlertDialog.FancyAlertButtonsClickListener {
+            override fun onPositiveButtonClick() {
+                handleDialogButtonClick(activity)
+            }
+
+            override fun onNegativeButtonClick() {
+                handleDialogButtonClick(activity)
+            }
+        }
```

Comment:
Fancy viewModel is removed in master. Simple dialogs are replaced with the `AdaptiveDialog` here, the rest use the click listener for now - we can transition them from `FancyAlertDialog` later.